lib/features/grid-snapping/behavior/ResizeBehavior.js

```diff
@@ -0,0 +1,183 @@
+import inherits from 'inherits';
+
+import CommandInterceptor from '../../../command/CommandInterceptor';
+
+import {
+  assign,
+  isString
+} from 'min-dash';
+
+
+/**
+ * Integrates resizing with grid snapping.
+ */
+export default function ResizeBehavior(eventBus, gridSnapping) {
+  CommandInterceptor.call(this, eventBus);
+
+  this._gridSnapping = gridSnapping;
+
+  var self = this;
+
+  this.preExecute('shape.resize', function(event) {
+    var context = event.context,
+        hints = context.hints || {},
+        autoResize = hints.autoResize;
+
+    if (!autoResize) {
+      return;
+    }
+
+    var shape = context.shape,
+        newBounds = context.newBounds;
+
+    if (isString(autoResize)) {
+      return self.snapComplex(shape, newBounds, autoResize);
+    }
+
+    self.snapSimple(shape, newBounds);

```

Comment:
Can we make the side effect (overriding newBounds) clear here? I.e.

```
newBounds = self.snapComplex(...);
```

Reply:
👍